src/tiled/mapeditor.cpp

```diff
@@ -673,6 +677,17 @@ void MapEditor::setWangFill(bool value)
     mBucketFillTool->setWangFill(value);
 }
 
+void MapEditor::extend(MapObjectItem *mapObjectItem, bool extendingFirst)
+{
+    setSelectedTool(mPolylineObjectsTool);
```

Comment:
Using this instead of `mToolManager->selectTool(mPolylineObjectsTool)` makes the tool switch without this being visible on the "Tools" tool bar. Was that a conscious decision? I think it wouldn't hurt for the user to see the active tool changing. Same for `extendingFinished`.